File: russian-doll-envelopes/russian-doll-envelopes.py
class Solution(object):
    def maxEnvelopes(self, nums):
        nums=sorted(nums,key=lambda i:(i[0],-i[1]))
        dp=[1 for i in range(len(nums))]
        for i in range(len(nums)):
            
            for j in range(i):
                if nums[i][1]>nums[j][1]:
                    dp[i]=max(dp[i],dp[j]+1)
        return max(dp)
        
        
        
        
        

File: russian-doll-envelopes/russian-doll-envelopes.cpp
class Solution {
public:
    int maxEnvelopes(vector<vector<int>>& nums) {
        sort(nums.begin(),nums.end(),[](vector <int> &a,vector <int> &b){return a[0]==b[0]?a[1]>b[1]:a[0]<b[0];});
        
        vector <int> dp;
        for(auto i:nums)
        {
            auto it=lower_bound(dp.begin(),dp.end(),i[1]);
            if(it==dp.end())
                dp.push_back(i[1]);
            else
                *it=i[1];
        }
        
        return dp.size();
        
        
    }
};